pkgs/development/compilers/gcc/common/configure-flags.nix

```diff
@@ -69,7 +69,6 @@ let
       "--with-as=${
         if targetPackages.stdenv.cc.bintools.isLLVM then binutils else targetPackages.stdenv.cc.bintools
       }/bin/${targetPlatform.config}-as"
-      "--with-ld=${targetPackages.stdenv.cc.bintools}/bin/${targetPlatform.config}-ld"
```

Comment:
`Another option would be to pass `"--with-ld=${targetPlatform.config}-ld"` and rely on `PATH` lookup.` (I hope it works as is).

Reply:
I don't think it would work. [Here](https://github.com/gcc-mirror/gcc/blob/1c5cf7cc82d3a2afa5d4fb90049456be408c2a7a/gcc/collect2.cc#L1056) and [here](https://github.com/gcc-mirror/gcc/blob/1c5cf7cc82d3a2afa5d4fb90049456be408c2a7a/gcc/gcc.cc#L3107) is the logic in GCC to check `DEFAULT_LINKER` (the `--with-ld` value). As you can see, it passes the value directly to `access` and doesn't search `$PATH`.

Reply:
Oh, yeah. Great find!